pytorch_lightning/accelerators/accelerator_connector.py

```diff
@@ -0,0 +1,473 @@
+                        raise MisconfigurationException(
+                            "Sharded Plugin is not supported with Apex AMP, "
+                            "please using native AMP for 16-bit precision."
+                        )
+                    log.info("Using APEX 16bit precision.")
+                    self.amp_type = AMPType.APEX
+                    return ApexMixedPrecisionPlugin(self.amp_level)
+        else:
+            raise NotImplementedError("We only support precisions 32 and 16!")
+
+    def select_training_type_plugin(self):
+        cluster_environment = self.select_cluster_environment()
+        if self.use_ddp2:
+            plugin = DDP2Plugin(parallel_devices=self.parallel_devices, cluster_environment=cluster_environment)
+        elif self.use_ddp:
+            use_slurm_ddp = self.use_ddp and self.is_slurm_managing_tasks
+            use_torchelastic_ddp = self.use_ddp and self.is_using_torchelastic
+            use_ddp_spawn = self._distrib_type == DistributedType.DDP_SPAWN
+            use_ddp_cpu_spawn = self.use_ddp and self.on_cpu
+            use_ddp_cpu_torch_elastic = use_ddp_cpu_spawn and self.is_using_torchelastic
+            use_ddp_cpu_slurm = use_ddp_cpu_spawn and self.is_slurm_managing_tasks
+            # use_ddp_sharded = self.distributed_backend == "ddp_sharded"
+            # use_ddp_sharded_spawn = self.distributed_backend == "ddp_sharded_spawn"
+
+            if self.on_tpu:
+                ddp_plugin_cls = TPUSpawnPlugin
+
+            # ddp script mode uses the same flags as TE
+            # TODO: decouple from TE
+            if os.environ.get("PL_IN_DDP_SUBPROCESS", False):
+                use_torchelastic_ddp = False
+
+            # fixme
+            # if use_ddp_sharded:
+            #     ddp_plugin_cls = DDPShardedPlugin
+            # elif use_ddp_sharded_spawn:
+            #     ddp_plugin_cls = DDPSpawnShardedPlugin
+            if use_ddp_cpu_slurm or use_slurm_ddp or use_ddp_cpu_torch_elastic or use_torchelastic_ddp:
+                ddp_plugin_cls = DDPPlugin
+            elif use_ddp_spawn or use_ddp_cpu_spawn:
+                ddp_plugin_cls = DDPSpawnPlugin
+            else:
+                ddp_plugin_cls = DDPPlugin
+
+            plugin = ddp_plugin_cls(
+                parallel_devices=self.parallel_devices,
+                num_nodes=self.num_nodes,
+                cluster_environment=cluster_environment,
+                sync_batchnorm=self.sync_batchnorm,
+            )
+        elif self.use_dp:
+            plugin = DataParallelPlugin(parallel_devices=self.parallel_devices)
+        elif self.use_horovod:
+            plugin = HorovodPlugin(parallel_devices=self.parallel_devices)
+        elif self.on_tpu:
+            plugin = SingleTPUPlugin(self.tpu_id)
+        else:
+            plugin = SingleDevicePlugin(device=torch.device(f"cuda:{self.root_gpu}" if self.on_gpu else "cpu"))
```

Comment:
Does the `SingleDevicePlugin`  needs to know about cpu or gpu ? Should this logic be in accelerator ?